src/profiler/profiler.h

```diff
@@ -1149,8 +1158,15 @@ struct ProfileOperator : public ProfileEvent {
     : ProfileEvent(name)
       , as_task_(name, &domain_)
       , name_(name)
-      , attributes_(attributes) {
-    SetCategories(domain_.name());
+      , attributes_(attributes)
+      , profiling_operator_(strcmp(name, "Dummy_Wait"))
+      , profiling_task_(strcmp(name, "Dummy_Wait") && strcmp(name, "Custom")) {
```

Comment:
"Dummy_Wait" is being used in multiple places. It's better to make this a constant string in the header file instead of plain string here.

Reply:
Having a constant string in custom-inl.h will introduce a circular dependency issue. I think we should keep it this way

Reply:
I also feel instead of comparing name to a const string, it might be better to write a function to explicitly call out the logic. `IsCustomOperator(const char *)`. Even for now, I already forgot what the comparison with a "Dummy_Wait" string is for :)